server/connector/connector.go

```diff
@@ -33,37 +39,37 @@ type Connector interface {
 type Conn struct {
 	Config  Config
 	Sender  Sender
+	Handler ResponseHandler
 	Manager Manager
 	Queue   Queue
 	Router  router.Router
 	KVStore kvstore.KVStore
-	Ctx     context.Context
-	Cancel  context.CancelFunc
+
+	Ctx    context.Context
+	Cancel context.CancelFunc
+
+	wg sync.WaitGroup
 }
 
 type Config struct {
 	Name       string
 	Schema     string
 	Prefix     string
 	URLPattern string
+	Workers    int
 }
 
-func NewConnector(router router.Router, sender Sender, config Config) (Conn, error) {
-	emptyConn := Conn{}
+func NewConnector(router router.Router, sender Sender, handler ResponseHandler, config Config) (*Conn, error) {
 	kvs, err := router.KVStore()
 	if err != nil {
-		return emptyConn, err
+		return nil, err
 	}
 
-	manager, err := NewManager(config.Schema, kvs)
-	if err != nil {
-		return emptyConn, err
-	}
-
-	return Conn{
+	return &Conn{
 		Config:  config,
 		Sender:  sender,
-		Manager: manager,
+		Manager: NewManager(config.Schema, kvs),
+		Queue:   NewQueue(sender, nil, config.Workers),
```

Comment:
alternatively, we can have a func on Queue: SetResponseHandler(...)
that we can use in the specific connectors APNS / FCM